composer/core/state.py

```diff
@@ -501,6 +525,19 @@ def algorithms(self):
     def algorithms(self, algorithms: Sequence[Algorithm]):
         self._algorithms[:] = algorithms
 
+    def _dataset_of(self, obj: Optional[Union[Iterable, Evaluator, DataSpec, DataLoader]]) -> Optional[Dataset]:
+        from composer.core.evaluator import Evaluator
+        if obj is None:
+            return None
+        if isinstance(obj, Evaluator):
+            obj = obj.dataloader
+        if isinstance(obj, DataSpec):
+            obj = obj.dataloader
+        if isinstance(obj, DataLoader):
+            return obj.dataset
```

Comment:
looks like this should be just `return obj` instead of `return obj.dataset` since `StreamingDataLoader` has a method `state_dict()` without any parameter. However, `StreamingDataset` has a method `state_dict()` with one parameter.

Reply:
Fixed that bug, and updated the doc/comments here.